phoenix-core/src/main/java/org/apache/phoenix/util/IndexUtil.java

```diff
@@ -958,6 +958,13 @@ public static void setScanAttributesForIndexReadRepair(Scan scan, PTable table,
             // This index table must be being deleted. No need to set the scan attributes
             return;
         }
+        // MetaDataClient modifies the index table name for view indexes if the parent view of an index has a child
+        // view. This, we need to recreate a PTable object with the correct table name for the rest of this code to work
+        if (table.getName().getString().contains(QueryConstants.CHILD_VIEW_INDEX_NAME_SEPARATOR)) {
```

Comment:
Should we add a check here that table is a view?
Can a user create a table with # in the name?

Reply:
I do not think it is allowed. Can you check it?

Reply:
This should check if table is view index..maybe you can use 'table.getViewIndexId() != null`?
I believe we disallow using '#' in user-defined names..you get:

```
java.lang.IllegalArgumentException: Illegal character code:35, <#> at 3. User-space table qualifiers can only contain 'alphanumeric characters': i.e. [a-zA-Z_0-9-.]:
```

Reply:
Just checked, yes it is not allowed for tables.

Reply:
Agree it would be better to not rely on a naming convention if possible to tell if it's a view index.

Reply:
Ok, I will add this extra check.

Reply:
Done